src/main/kotlin/com/bloock/sdk/record/entity/document/Document.kt

```diff
@@ -0,0 +1,94 @@
+package com.bloock.sdk.record.entity.document
+
+import com.bloock.sdk.proof.entity.Proof
+import com.bloock.sdk.shared.Signature
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+
+abstract class Document<T>(src: T, args: DocumentLoadArgs) {
+    var ready: Unit
```

Comment:
I think you don't need the ready parameter in Java/Kotlin as you are treating everything synchronously (runBlocking), so you can remove it.